addons/website/static/src/js/menu/new_content.js

```diff
@@ -65,23 +109,181 @@ var NewContentMenu = websiteNavbarData.WebsiteNavbarActionWidget.extend({
     //--------------------------------------------------------------------------
 
     /**
-     * Called when the menu's toggle button is clicked -> Opens the menu.
+     * Called when the menu's toggle button is clicked:
+     *  -> Opens the menu and reset the tab navigation (if closed)
+     *  -> Close the menu (if open)
+     * Called when a click outside the menu's options occurs -> Close the menu
      *
      * @private
      * @param {Event} ev
      */
-    _onMenuToggleClick: function (ev) {
-        ev.preventDefault();
-        this.$newContentMenuChoices.toggleClass('o_hidden');
+    _onBackgroundClick: function (ev) {
+        // condition to prevent a crash when clicking on the button before the start is finished
+        if (this.$newContentMenuChoices) {
+            if (this.$newContentMenuChoices.hasClass('o_hidden')) {
+                this._showMenu();
+            } else {
+                this._hideMenu();
+            }
+        }
     },
     /**
-     * Called when a click outside the menu's options occurs -> Closes the menu
+     * Called when a keydown occurs:
+     *  ESC -> Closes the modal
+     *  TAB -> Navigation (captured in the modal)
      *
      * @private
      * @param {Event} ev
      */
-    _onBackgroundClick: function (ev) {
+    _onBackgroundKeydown: function (ev) {
+        switch (ev.which) {
+            case $.ui.keyCode.ESCAPE:
+                this._hideMenu();
+                break;
+            case $.ui.keyCode.TAB:
+                if (ev.shiftKey) {
+                    if (this.firstTab || document.activeElement === this.$firstLink[0]) {
+                        this._focusLastLink();
+                        ev.preventDefault();
+                    }
+                } else {
+                    if (this.firstTab || document.activeElement === this.$lastLink[0]) {
+                        this._focusFirstLink();
+                        ev.preventDefault();
+                    }
+                }
+                this.firstTab = false;
```

Comment:
As discussed, we want to capture the navigation inside the popup, because it doesn't make sense to be able to select elements that are potentially "hidden" below. This is the same behavior as bs4 modal.

If you decided to click on "new" and wanted to navigate to "edit" instead, you can just press esc first and then tab.